channel.go

```diff
@@ -315,6 +310,14 @@ func (ch *Channel) UnBanUser(targetID string, options map[string]string) error {
 	return ch.client.UnBanUser(targetID, options)
 }
 
+// newChannel initializes channel with required private fields
```

Comment:
Include the required fields (like ID) and dedupe (with a sync.Map LoadOrStore). This will prevent bugs on the client side.

Reply:
it's not public method; what means dedupe?

Reply:
Users can instantiate the same channel twice with the API. A simple deduplication check in combination with your thread safe implementation ensures updates are apply synchronously.